### DISCUSSION TITLE
Should we change the namespace?

### DISCUSSION THREAD

**Discussion by timunie** [PieroCastillo/Aura.UI#29] [General]
Hi @PieroCastillo ,

I often find my self wondering if a control is from Aura.UI or Avalonia, as both libs share a common namespace. Moreover this may lead to issues in the future: 
- What if Avalonia adds a new Control with the same name, e.g. a `NavigationView`? 
- What if another lib also shares this namespace?

Beside this I think if one works with several programmers on the same app and you have the namespace like this: 
```xaml
xmlns:aura="https://github.com/PieroCastillo/Aura.UI"
```
anybody can just open the link and see the docs and knows where the origin of the controls is 😄 . 

What do you think about this?

**NOTE:** I know that this is a breaking change and will affect many many other programs. I wanted to share my idea anyway. 

Happy coding
Tim

**Comment by PieroCastillo** [comment]
u right, this is a very big breaking change, in fact this problem already occurs with Arc Control, but I don't know if that is a good idea

**Comment by PieroCastillo** [comment]
I think if that occurs I can rename the controls, just adding "Aura" to its name (like AuraTabView or AuraTabItem, in the case of some Avalonia NavigationView I can rename it as "AuraNavigationView")

**Reply by timunie** [reply]
Yeah but this also has some drawbacks:

- if you don't have the time to do the renaming for whatever reason, the issue is unsolved for a very long time.
- also renaming is a breaking change. So in that case breaking changes may occur several times, which is also not good
- last but not least one may even not notice that the control was renamed and is wondering over a strange behavior. 

I think there is an option to create a namespace in your own App which maps the old one, so with little code the breaking change is overcome. Haven't tested it yet.

But for sure these are just my two cents. 

Have a great weekend
Tim

**Reply by PieroCastillo** [reply]
maybe... well, I think I can do it, also, it is easy to implement, well I'll implement it

**Reply by timunie** [reply]
I didn't want to force you in either direction. Just wanted to give some feedback and thoughts from a user perspective. So if you are not happy with the idea keep Aura.UI as it is. 😊

**Reply by timunie** [reply]
One last thing: If you want to change it, I can do and provide a PR. Just give me a hint.

**Reply by PieroCastillo** [reply]
sorry, but I don't want to change it, I hope you will understand.

**Reply by timunie** [reply]
Understood, thank you for the info, 👍
